src/components/vm/vmActions.tsx

```diff
@@ -44,164 +44,134 @@ import { RenameDialog } from './vmRenameDialog.jsx';
-                    detail: ex.message,
-                }
-            })
-        );
-    })
-);
+export async function vmShutdown(vm: VM) {
+    startOperationProgress(vm);
+    try {
+        await vmDomainMethod<void>(vm, 'Shutdown', 'u', 0);
+        if (!vm.persistent)
+            cockpit.location.go(["vms"]);
+    } catch (ex) {
+        setVmError(vm, cockpit.format(_("VM $0 failed to shutdown"), vm.name), ex);
+    }
+}
 
-const onResume = (vm: VM) => (
-    domainResume({ id: vm.id, connectionName: vm.connectionName }).catch(ex => {
-        store.dispatch(
-            updateVm({
-                connectionName: vm.connectionName,
-                name: vm.name,
-                error: {
-                    text: cockpit.format(_("VM $0 failed to resume"), vm.name),
-                    detail: ex.message,
-                }
-            })
-        );
-    })
-);
-
-const onForceoff = (vm: VM) => (
-    domainForceOff({ id: vm.id, connectionName: vm.connectionName })
-            .then(() => !vm.persistent && cockpit.location.go(["vms"]))
-            .catch(ex => {
-                store.dispatch(
-                    updateVm({
-                        connectionName: vm.connectionName,
-                        name: vm.name,
-                        error: {
-                            text: cockpit.format(_("VM $0 failed to force shutdown"), vm.name),
-                            detail: ex.message,
-                        }
-                    })
-                );
-            })
-);
+export async function vmPause(vm: VM) {
+    try {
+        await vmDomainMethod<void>(vm, 'Suspend', '');
+    } catch (ex) {
+        setVmError(vm, cockpit.format(_("VM $0 failed to pause"), vm.name), ex);
+    }
+}
 
-const onSendNMI = (vm: VM) => (
-    domainSendNMI({ id: vm.id, connectionName: vm.connectionName }).catch(ex => {
-        store.dispatch(
-            updateVm({
-                connectionName: vm.connectionName,
-                name: vm.name,
-                error: {
-                    text: cockpit.format(_("VM $0 failed to send NMI"), vm.name),
-                    detail: ex.message,
-                }
-            })
-        );
-    })
-);
+export async function vmResume(vm: VM) {
+    try {
+        await vmDomainMethod<void>(vm, 'Resume', '');
+    } catch (ex) {
+        setVmError(vm, cockpit.format(_("VM $0 failed to resume"), vm.name), ex);
+    }
+}
+
+export async function vmForceOff(vm: VM) {
+    try {
+        await vmDomainMethod<void>(vm, 'Destroy', 'u', 0);
+        if (!vm.persistent)
+            cockpit.location.go(["vms"]);
+    } catch (ex) {
+        setVmError(vm, cockpit.format(_("VM $0 failed to force shutdown"), vm.name), ex);
+    }
+}
+
+export async function onSendNMI(vm: VM) {
+    try {
+        await vmDomainMethod<void>(vm, 'InjectNMI', 'u', 0);
+    } catch (ex) {
+        setVmError(vm, cockpit.format(_("VM $0 failed to send NMI"), vm.name), ex);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2376-46b24e73-20251028-114421-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)